File: react-native/react-native-pulsar/deps/Pulsar/Sources/Pulsar/Pulsar.swift
import CoreHaptics
import UIKit

@objc public class Pulsar: NSObject {
  private var engine = HapticEngineWrapper()
  private var audioSimulator = AudioSimulator()
  private var presets: PresetsWrapper?
  private var realtimeComposer: RealtimeComposer!
  
  @objc public override init() {
    realtimeComposer = RealtimeComposer(engine: self.engine)
  }
  
  @objc public func getPresets() -> PresetsWrapper {
    if (presets == nil) {
      presets = PresetsWrapper(haptics: self)
    }
    return presets!
  }
  
  @objc public func preloadPresets(presetNames: Array<String>) {
    let presets = self.getPresets()
    for (presetName) in presetNames {
      presets.preloadPresetByName(presetName)
    }
  }
  
  @objc public func enableSound(state: Bool) {
    audioSimulator.enableSound(state)
  }
  
  @objc public func enableCache(state: Bool) {
    self.getPresets().enableCache(state: state)
  }
  
  @objc public func clearCache() {
    self.getPresets().resetCache()
  }
  
  @objc public func getPatternComposer() -> PatternComposer {
    return PatternComposer(engine: engine, audioSimulator: audioSimulator)
  }
  
  @objc public func getRealtimeComposer() -> RealtimeComposer {
    return realtimeComposer
  }
  
}

File: react-native/react-native-pulsar/android/src/main/java/com/swmansion/pulsar/Pulsar.kt
package com.swmansion.pulsar

import android.content.Context
import com.swmansion.pulsar.audio.AudioSimulator
import com.swmansion.pulsar.composers.PatternComposer
import com.swmansion.pulsar.composers.RealtimeComposer
import com.swmansion.pulsar.haptics.HapticEngineWrapper
import com.swmansion.pulsar.presets.PresetsWrapper
import com.swmansion.pulsar.types.CompatibilityMode
import com.swmansion.pulsar.types.RealtimeComposerStrategy

class Pulsar(context: Context) {
    private val engine = HapticEngineWrapper(context)
    private val audioSimulator = AudioSimulator(engine.getRealCompatibilityMode())
    private var presets: PresetsWrapper? = null
    private var realtimeComposer: RealtimeComposer? = null

    fun getPresets(): PresetsWrapper {
        if (presets == null) {
            presets = PresetsWrapper(this)
        }
        return presets!!
    }

    fun preloadPresets(presetNames: List<String>) {
        val presets = this.getPresets()
        for (presetName in presetNames) {
            presets.preloadPresetByName(presetName)
        }
    }

    fun enableSound(state: Boolean) {
        audioSimulator.enableSound(state)
    }

    fun enableCache(state: Boolean) {
        this.getPresets().enableCache(state)
    }

    fun clearCache() {
        this.getPresets().resetCache()
    }

    fun getPatternComposer(): PatternComposer {
        return PatternComposer(engine, audioSimulator)
    }

    fun getRealtimeComposer(strategy: RealtimeComposerStrategy = RealtimeComposerStrategy.ENVELOPE): RealtimeComposer {
        if (realtimeComposer == null) {
            realtimeComposer = RealtimeComposer(engine, strategy)
        }
        return realtimeComposer!!
    }

    fun hapticSupport(): CompatibilityMode {
        return engine.getRealCompatibilityMode()
    }

    fun simulateCompatibilityMode(mode: CompatibilityMode) {
        engine.simulateCompatibilityMode(mode)
        audioSimulator.setCompatibilityMode(mode)
    }
}